packages/core/parcel-bundler/test/utils.js

```diff
@@ -272,6 +272,25 @@ function normaliseNewlines(text) {
   return text.replace(/(\r\n|\n|\r)/g, '\n');
 }
 
+function packageInstall(dir) {
```

Comment:
Why is this a test util? Parcel already has autoinstall

Reply:
@devongovett I've tried without installing the package before starting the test. But, somehow it fails to recognise that it needs to do installation in the root directory rather than the package itself:

Here is the scenario:
- Yarn workspace example has three packages: `package-a`, `package-b` and `package-main`.
- I am running parcel on `package-main` package (given root directory set to yarn workspace root directory)

`package-main` has a dependency to `package-a`. Parcel recognises that `package-main` is part of `yarn` ecosystem (by traversing the tree down to `yarn.lock` file in the root). However, for some reason, it runs `yarn add package-a` inside `package-main`! Any ideas?

I will look further into this.

Reply:
@devongovett Could you help me on this? :)

Do we expect the developers to run `npm install` or `yarn` when they start working on a brand new project they just pulled from a repository? Or do we expect them to run `parcel` right away, and that is supposed to work?

What's happening now is that it tries to install each dependency individually if they are not installed. Isn't that incredibly slow if you have hundreds of dependencies (in a large project code base, which is likely in a yarn workspace projects)?

I was thinking, if this is the path for Parcel, we can still make it happen. I just wanted to make sure we are on the same page. 

The solution I have in mind is somehow modify `resolvePackageAliases` where it checks not only `source`, `alias`, and `browser`, but also `workspaces`. The downside is that to resolve a single dependency, we need to read all `package.json` files defined in `package.json/workspaces` packages, resolve their names and see if the any matches the dependency. We can cache this action, but then we need to think about how we can bust the cache in case the user adds another package folder.